aexpect/ops.py

```diff
@@ -0,0 +1,479 @@
+# file ops
+###############################################################################
+
+
+def ls(session, path, quote_path=True, flags="-1UNq"):  # pylint: disable=C0103
+    """
+    Run `ls` in given directory through a session.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str path: absolute or relative path to list
+    :param bool quote_path: whether to quote (escape) or not (expand) the path
+    :param str flags: raw flags to pass to ls command
+    :returns: names of files in dir (without path components)
+    :rtype: [str]
+
+    Just like :py:func:`os.listdir`, does not include file names starting with
+    dot (`'.'`)
+    """
+    cmd = f'ls {flags} {quote(path)}' if quote_path else f'ls {flags} {path}'
+    status, output = session.cmd_status_output(cmd)
+    if output:
+        output = output.strip()
+    if status == 2:      # probably just nothing found
+        return []
+    if status != 0:
+        raise RuntimeError(f'Failed to ls {path}: {output}')
+    return output.splitlines()
+
+
+def move(session, source, target, quote_path=True, flags=""):
+    """
+    Move file or directory from source to target / rename source to target.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str source: full path of existing file or directory
+    :param str target: full path of new file or directory
+    :param bool quote_path: whether to quote (escape) or not (expand) the path
+    :param str flags: raw flags to pass to mv command
+    :raises: :py:class:`RuntimeError` if mv returns non-zero exit status
+
+    Calls `mv source target` through a session. See `man mv` for what source
+    and target can be and what behavior to expect.
+    """
+    cmd = f'mv {flags} {quote(source)} {quote(target)}' if quote_path \
+        else f'mv {flags} {source} {target}'
+    status, output = session.cmd_status_output(cmd)
+    if status != 0:
+        raise RuntimeError(f"Failed to move {source} to {target}: "
+                           f"{output.strip() if output else ''}")
+
+
+def copy(session, source, target, quote_path=True, flags=""):
+    """
+    Copy file or directory from source to target.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str source: full path of existing file or directory
+    :param str target: full path of new file or directory
+    :param bool quote_path: whether to quote (escape) or not (expand) the path
+    :param str flags: raw flags to pass to cp command
+    :raises: :py:class:`RuntimeError` if cp returns non-zero exit status
+
+    Calls `cp source target` through a session. See `man cp` for what source
+    and target can be and what behavior to expect.
+    """
+    cmd = f'cp {flags} {quote(source)} {quote(target)}' if quote_path \
+        else f'cp {flags} {source} {target}'
```

Comment:
Wouldn't it be easier to follow with pure old if? (I'm not a fun of backslashed long expressions)

Reply:
I also don't like backslashed expressions but having one liner if statements fits better when defining a single value so at least on my side I would prefer to keep it as is. Tell me if you insist on this change, the current statement seems readable and shorter than full blown if clauses to me and might even be refactored through some standardization on the quote behaviors in commands later on.

Reply:
It can stay as is although I disagree with the readability, but it's just a matter of taste :-)

Reply:
Unless one could calculate the entropy of the two statements and prove your version contains more information and/or correlates the thought invocations when parsing the line with the typical neural pathways of a human reader ^^